### PR TITLE
23-徐晔-HW2

### PR DESCRIPTION
**Pull request by yexu15**

### CODE REVIEW

File: Lesson2/assignment/yours.sol [hunk 1]
```diff
@@ -1 +1,88 @@
...
+        employee.id.transfer(payment);
+    }
+
+    function _findEmployee(address employeeId) private returns (Employee, uint) {
+        for (uint i = 0; i < employees.length; i++){
+            if (employees[i].id == employeeId){
+                return (employees[i], i);
+            }
+        }
+    }
```
Comment (jonnyhsy) [inline Lesson2/assignment/yours.sol]: 编译器也许会优化返回返回类型的默认值， 但逻辑完整性应该要加上 return (Employee(0,0,0), 0);

File: Lesson2/assignment/yours.sol [hunk 2]
```diff
@@ -1 +1,88 @@
...
+
+    function updateEmployee(address employeeId, uint salary) {
+        require(msg.sender == owner);
+        var (employee, index)  = _findEmployee(employeeId);
+        assert(employee.id != 0x0);
+        _partialPaid(employee);
+        current_totalSalary += salary * 1 ether - employees[index].salary;
+        employees[index].salary = salary * 1 ether;
+        employees[index].lastPayday = now;
```
Comment (jonnyhsy) [inline Lesson2/assignment/yours.sol]: 可以简写成 employees[index] = Employee(employeeId, salary * 1 ether, now);

**Comment by yexu15** [comment]
Thanks @jonnyhsy! I just refined the code based on your suggestion. 
One question, when I wrote code as blew, I got the error (expected identifier got 'lparen') , I think I have to put "employees.push(0, 0, 0);" into a function, but I am not sure why. 


contract Payroll {
    struct Employee {
        address id;
        uint salary;
        uint lastPayday;
    }
    Employee [] employees;
    employees.push(0, 0, 0);
}

**Comment by jonnyhsy** [comment]
I think you need to do employees.push(Employee(0,0,0)) instead of employees.push(0,0,0);  yeah this need to happen inside a function as it only defines states in the contract context.

**Pull request merged**